test/e2e_node/gci-init.sh

```diff
@@ -24,25 +24,23 @@ if [ -f "${CONTAINERD_HOME}/${CONTAINERD_ENV_METADATA}" ]; then
   source "${CONTAINERD_HOME}/${CONTAINERD_ENV_METADATA}"
 fi
 
-if [ "${CONTAINERD_CGROUPV2:-"false"}"  == "true" ]; then
-  # check cos image
-  if [ -r /etc/os-release ]; then
-    OS_ID="$(. /etc/os-release && echo "$ID")"
-  fi
-  if [ "${OS_ID}" = "cos" ]; then
-    if ! grep -q 'systemd.unified_cgroup_hierarchy=true' /proc/cmdline && [ "$(stat -fc %T /sys/fs/cgroup/)" != "cgroup2fs" ]; then
-      echo "Setting up cgroupv2"
-
-      mount_path="/tmp/esp"
-      mkdir -p "${mount_path}"
-      esp_partition="/dev/sda12"
-      mount "${esp_partition}" "${mount_path}"
-      sed -i 's/systemd.unified_cgroup_hierarchy=false/systemd.unified_cgroup_hierarchy=true/g' "${mount_path}/efi/boot/grub.cfg"
-      umount "${mount_path}"
-      rmdir "${mount_path}"
-
-      echo "Reconfigured grub; rebooting..."
-      reboot
+if [[ -n "${CONTAINERD_COS_CGROUP_MODE}" ]]; then
```

Comment:
I'm not sure how this script is invoked, but it might be easier to read if the prerequisite conditionals are set up so the script can exit early.

For example:

```bash
[[ -n "${CONTAINERD_COS_CGROUP_MODE}" ]] || exit 0
[[ -r /etc/os-release ]] || exit 0
OS_ID="$(cat /etc/os-release | grep '^ID=' | sed -e 's/ID=//')"
[[ "${OS_ID}" = "cos" ]] || exit 0
[[ $(command -v cgroup_helper >/dev/null 2>&1) ]] || exit 0
```

and then the rest of your actual logic can follow.

Reply:
Thanks for the suggestion. I agree it would be nice too refactor to exit early if  prerequisite conditionals are not met. I can't `exit 0` since this script is run as part of cloud-init and we still want it to run the later parts of the script even if some of the cgroup mode checks fail.

As a result, I refactored it into a bash function so I can return early from the function if needed. I think it's much cleaner that way, PTAL, thanks for the suggestion!

Reply:
Yep, I wasn't sure how it was invoked and whether `exit 0` would have been appropriate.  A function works well too.